fastlane/Fastfile

```diff
@@ -829,15 +824,36 @@ end
       type: "appstore",
       team_id: get_required_env("EXT_EXPORT_TEAM_ID"),
       readonly: options[:readonly] || is_ci,
-      app_identifier: [
-        "com.codality.NotationalFlow",
-        "com.codality.NotationalFlow.Share",
-        "com.codality.NotationalFlow.Widgets",
-      ],
+      app_identifier: simplenote_app_identifiers,
       template_name: "NotationalFlow Keychain Access (Distribution)",
     )
   end
 
+  # Compiles the array of bundle identifiers for the different targets that
+  # make up the Simplenote app, to be used as the `app_identifier` parameter
+  # for the `match` action.
+  def simplenote_app_identifiers(root_bundle_id: APP_STORE_BUNDLE_IDENTIFIER)
+    extension_bundle_ids =
+      %w[Share Widgets Intents]
+      .map { |suffix| "#{root_bundle_id}.#{suffix}" }
+
+    [root_bundle_id, *extension_bundle_ids]
+  end
+
+  # Compiles the dictionary mapping of the bundle identifiers and provisioning
+  # profiles to be used in the `export_options > provisioningProfiles`
+  # parameter for the `build_app`/`gym` action.
+  def simplenote_provisioning_profiles(root_bundle_id: APP_STORE_BUNDLE_IDENTIFIER, match_type: 'AppStore')
```

Comment:
Leaving the caller the ability to specify the match type (`AppStore`, `InHouse`, etc) as a String leaves the door open for typos and use of unsupported types (maybe during a migration to a new major version of the tool?)

I decided to stick with this approach for now to avoid adding extra complexity to the code here by defining something like a symbol for the type and then switching on its value.

Definitely open to the idea of doing it, but maybe in a followup step or as a more abstract method in the release toolkit.

Reply:
I think this is fine to keep it that way especially because this is an internal function and not a lane, so that value will only be provided by callers elsewhere in the `Fastfile`.

Alternatively you could make this parameter be a boolean `in_house: false` and use `match_type = in_house ? 'InHouse' : 'AppStore'` in the implementation. Or even not have that parameter at all, and ise `match_type = root_bundle_id == APP_STORE_BUNDLE_IDENTIFIER ? 'AppStore' : 'InHouse' (though maybe it's taking it too far?)  🤷